lib/oeqa/selftest/cases/updater_qemux86_64.py

```diff
@@ -204,6 +207,26 @@ def test_provisioning(self):
         verifyProvisioned(self, machine)
 
 
+class DeviceCredProvTestsNonOSTree(DeviceCredProvTests):
+    def setUpLocal(self):
+        global UBOOT_ENABLE
+        UBOOT_ENABLE = 'no'
```

Comment:
Do we have to treat this as a global?

Reply:
Yes, in another case we will get a warning like ```Shadows name ...```

Reply:
I was wondering if it could be passed as an argument like we do for other qemu configuration options.

Reply:
By default, this argument is set to ```yes```, only in case we want to run ```qemu``` without  ```u-boot``` we change it, and of course, at the end of the test we just restore it to default state, that's actually the whole idea. We have to have since we inherit from the test which is originally designed for the ```ostree``` case.

Reply:
Could we make a member variable of the `SharedCredProvTests`, overwrite it in this test's `setUpLocal` before calling the parent class, and then pass it to `qemu_launch` in the parent class function?

I'm concerned about the global variable having unintended side effects in the future.

Reply:
I just realized that python unit tests are not designed to be used with inheritance mechanism, I'm trying to define class member var to use it across class instances, so where is the proper place to initialize class self var? 
The idea is that I want to initialize ```self.VAR``` before invoking ```setUpLocal(self)``` member func, the reason why I want to do that is that I inherit my test class from the existing one and I want to have ```self.VAR``` which I'll be able to initialize before invoking ```setUpLocal(self)``` from the parent class.

I can't use ```__init__ ```, so when I try to use ```setUp(self)``` like
```
setUp(self):
  self.uboot_enable = 'no'
```
...all others self.VAR's declared in the parent class become invisible in the subclass?

Reply:
> I just realized that python unit tests are not designed to be used with inheritance mechanism

Oh, strange. What's different about python unit tests where inheritance doesn't work? I do seem to recall that this `setUpLocal` function is basically the class initializer, but I seem to also recall from the Yocto docs that there was some other `setUpXyz` function that might be helpful here.

If necessary, you could try setting the variable in the subclass, and then in the superclass, check if it is empty and set it to the real "default" if it is.

Reply:
The issue was that I call `super().setUpLocal` I did reworking over that, could you check out the latest changes?

Reply:
I wonder how it affects but if I override` setUpLocal()` or `setUp` all `self.VAR's `from the parent class are not available in the derived class...

Reply:
>  I did reworking over that, could you check out the latest changes?

Yep, looks fine to me now! If CI passes, I think this is ready to merge.